instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/EVRerouteTest.kt

```diff
@@ -276,16 +297,144 @@ class EVRerouteTest : BaseTest<EmptyTestActivity>(EmptyTestActivity::class.java)
                 KEY_EV_PRECONDITIONING_TIME to preconditioningTime,
             )
         )
+        checkDoesNotHaveParameters(urlAfterEmptyUpdate, userProvidedCpoiKeys)
+    }
+
+    @Ignore("Unignore when server is ready")
+    @Test
+    fun ev_reroute_parameters_with_user_provided_cpoi_ids() = sdkTest {
+        val stationIds = ";ocm-176357;ocm-190632"
+        val threeCoordinates = listOf(
+            Point.fromLngLat(11.585226, 48.176099),
+            Point.fromLngLat(11.063842, 48.39023),
+            Point.fromLngLat(10.32645, 49.069138),
+        )
+
+        mockWebServerRule.requestHandlers.clear()
+        routeHandler = MockDirectionsRequestHandler(
+            "driving-traffic",
+            readRawFileText(activity, R.raw.ev_route_response_custom_station_ids),
+            threeCoordinates,
+            relaxedExpectedCoordinates = true
+        )
+        mockWebServerRule.requestHandlers.add(routeHandler)
+
+        val requestedRoutes = requestRoutes(threeCoordinates, electric = true, stationIds = stationIds)
+
+        mapboxNavigation.startTripSession()
+        stayOnInitialPosition()
+        mapboxNavigation.setNavigationRoutesAndWaitForUpdate(requestedRoutes)
+        stayOnPosition(offRouteLocationUpdate.latitude, offRouteLocationUpdate.longitude)
+        waitForReroute()
+
+        val url1 = routeHandler.handledRequests.last().requestUrl!!
+        checkHasParameters(
+            url1,
+            mapOf(
+                KEY_ENGINE to VALUE_ELECTRIC,
+                KEY_WAYPOINTS_STATION_ID to stationIds
+            )
+        )
+
+        val offRouteLocationUpdate2 = mockLocationUpdatesRule.generateLocationUpdate {
+            latitude = offRouteLocationUpdate.latitude + 0.002
+            longitude = offRouteLocationUpdate.longitude
+        }
+        stayOnPosition(offRouteLocationUpdate2.latitude, offRouteLocationUpdate2.longitude)
+
+        waitForNewReroute()
+
+        val url2 = routeHandler.handledRequests.last().requestUrl!!
+        checkHasParameters(
+            url2,
+            mapOf(
+                KEY_ENGINE to VALUE_ELECTRIC,
+                KEY_WAYPOINTS_STATION_ID to stationIds
+            )
+        )
+    }
+
```

Comment:
Not sure if it's necessary, but maybe it's also worth checking reroute on the second leg?

Reply:
Oooh, you discovered a bug, yes, it's necessary!